tests/integration/release/utilities.py

```diff
@@ -359,3 +397,122 @@ def run_guardian_duplication_tests(
                 target_proportion=expected_noise,
                 name_additional="noised_data",
             )
+
+
+def get_guardian_duplication_info(
+    data: pd.DataFrame,
+) -> pd.DataFrame:
+    """Takes a list of prenoised DataFrames and matching noised Datasets and checks that
+    after applying duplicate_with_guardian, 1) no simulants are duplicated more than once,
+    2) address information was duplicated correctly and 3) the expected number of simulants
+    were duplicated."""
+    numerators = {
+        Keys.ROW_PROBABILITY_IN_HOUSEHOLDS_UNDER_18: 0,
+        Keys.ROW_PROBABILITY_IN_COLLEGE_GROUP_QUARTERS_UNDER_24: 0,
+    }
+    denominators = {
+        Keys.ROW_PROBABILITY_IN_HOUSEHOLDS_UNDER_18: 0,
+        Keys.ROW_PROBABILITY_IN_COLLEGE_GROUP_QUARTERS_UNDER_24: 0,
+    }
+    noised_data = get_noised_data(data)
+    prenoised_data = get_prenoised_data(data)
+    prenoised_data.columns = noised_data.columns
+
+    # get duplicated simulants
+    duplicated = noised_data.loc[noised_data["simulant_id"].duplicated()]
+
+    # add old housing type data to duplicated simulants
+    old_housing_data = prenoised_data[["simulant_id", "housing_type"]].rename(
+        {"housing_type": "unnoised_housing_type"}, axis=1
+    )
+    duplicated = duplicated.merge(old_housing_data)
+
+    # collect data for fuzzy checking for household under 18 and for college under 24
+    for probability_name, max_age, housing_type in zip(
+        [
+            Keys.ROW_PROBABILITY_IN_HOUSEHOLDS_UNDER_18,
+            Keys.ROW_PROBABILITY_IN_COLLEGE_GROUP_QUARTERS_UNDER_24,
+        ],
+        [18, 24],
+        ["Household", "College"],
+    ):
+        group_data = prenoised_data.loc[
+            (prenoised_data["age"] < max_age)
+            & (prenoised_data["housing_type"] == housing_type)
+            & (prenoised_data["guardian_1"].notna())
+        ]
+        merged_data = merge_dependents_and_guardians(group_data, prenoised_data)
+        sims_eligible_for_duplication = merged_data.index[
+            (
+                (merged_data["household_id"] != merged_data["guardian_1_household_id"])
+                & (merged_data["guardian_1_household_id"].notna())
+            )
+            | (
+                (merged_data["household_id"] != merged_data["guardian_2_household_id"])
+                & (merged_data["guardian_2_household_id"].notna())
+            )
+        ]
+        duplicated_in_group = duplicated.loc[
+            (duplicated["age"] < max_age)
+            & (duplicated["unnoised_housing_type"] == housing_type)
+        ]
+
+        numerators[probability_name] += len(duplicated_in_group)
+        denominators[probability_name] += len(sims_eligible_for_duplication)
+
+    # Only duplicate a dependent one time
+    has_multiple_duplications = noised_data["simulant_id"].value_counts().max() > 2
```

Comment:
This has flipped from our original check that no simulants were duplicated more than once. We instead want this to be True if any simulants were duplicated more than once, rather than none.